c2corg_ui/static/js/addassociation.js

```diff
@@ -65,11 +83,23 @@ app.AddAssociationController = function(appApi, $attrs) {
 
 
 /**
+ * Associate two documents. Possible associations are listed under http://bit.ly/1T15jxx
+ * For example, we cannot directly associate a waypoint to a route, but a route to a waypoint
+ * is possible, so we have to do it the other way around and change parent/child IDs accordingly. Same for outings.
  * @param {appx.SearchDocument} doc Document
  * @export
  */
 app.AddAssociationController.prototype.associate = function(doc) {
-  this.api_.associateDocument(this.parentId, doc).then(function() {
+  var parentId, childId;
+  // if the parentDoc is a Route and the childDoc is a Waypoint OR the parentDoc is an Outing, inverse the IDs.
+  if ((this.parentDocType === 'routes' && doc['waypoint_type']) || this.parentDocType === 'outings') {
```

Comment:
I think a ``doctype`` attribute is now available. That would clearer to use it (``doc['doctype'] === 'waypoints'``) instead of  ``doc['waypoint_type']`` to check that the current document is a WP.